### ISSUE TITLE
Report mentor email address, CYA and submit

### ISSUE THREAD

**Issue by mason-emily** [DFE-Digital/register-early-career-teachers#769]
## Description
This story delivers further steps in the journey for assigning and registering a new mentor for an ECT, including entering the mentor's email, checking your answers and submitting to reach the confirmation page.

## User story

**As** school user
**I need** to report the mentor's email address and check the details I've entered before submitting
**So that** DfE have the correct details they need to enable mentoring and/or mentor training

## Prototype, screenshots & content (p: ecf2)
[Prototype email address page](https://ecf2-prototype-e3c7aba62d82.herokuapp.com/v5/school/mentor/email-address)
[Prototype CYA & confirmation page](https://ecf2-prototype-e3c7aba62d82.herokuapp.com/v5/school/mentor/check-answers)

Note, their are additional pages in the prototype for collecting training details which are **not** in scope for this ticket.

**Screenshots of relevant screens:**

![image](https://github.com/user-attachments/assets/aa3ef8fb-5c16-497b-8ba3-b4fb32c49e1f)

Content should be as per the prototype above.
URL and validation messages should be as per [this doc](https://educationgovuk.sharepoint.com/:x:/r/sites/TeacherServices/Shared%20Documents/Teacher%20Continuing%20Professional%20Development/Teacher%20CPD%20Team/Register%20early%20career%20teachers/Beta/Design/Register%20ECTs%20page%20titles%20and%20URLs.xlsx?d=w0c1578ea643147b29357c7f610ee178f&csf=1&web=1&e=U8iIdi).

## Acceptance criteria

- [ ] 1. When user clicks 'Confirm and continue' on the <ins>'Check mentor details'</ins> page (#759), they will be navigated to the <ins>'What is [mentor name]'s email address?'</ins> page.
- [ ] 2. On the <ins>'What is [mentor name]'s email address?'</ins> page:
    - [ ] 2a. The H1 question will pull in the mentor's name e.g. What is [Samuel Taylor]'s email address?
    - [ ] 2b. User can enter an email address. See [here](https://educationgovuk.sharepoint.com/:x:/r/sites/TeacherServices/Shared%20Documents/Teacher%20Continuing%20Professional%20Development/Teacher%20CPD%20Team/Register%20early%20career%20teachers/Beta/Design/Register%20ECTs%20page%20titles%20and%20URLs.xlsx?d=w0c1578ea643147b29357c7f610ee178f&csf=1&web=1&e=CHF69R&nav=MTVfezk0RjM1RTQ2LUM5MjMtNEQ3Ny1BREQ2LTQzNjY0Q0I1QjAzRn0) for validation rules and error message content for the email field, which will be the same as for the ECT journey.
    - [ ] 2c.. When user clicks 'Continue' on the <ins>'What is [mentor name]'s email address?'</ins> page, they will be navigated to the <ins>'Check your answers before registering [mentor name] as a mentor'</ins> page. Note, there are additional pages in the prototype that are not within scope for now.
    - [ ] 2d. There will be a 'Back' link on the page that takes the user back to the <ins>'Check mentor details'</ins> page.
- [ ] 3. On the <ins>'Check your answers before registering [mentor name] as a mentor'</ins> page:
    - [ ] 3a. The mentor's TRN, name and email address will be shown. Note, there are additional rows and 'Change' links in the prototype which are not in scope for this ticket.
    - [ ] 3b. There will be a line to say the ECT this mentor is being assigned to (e.g. Samuel Taylor will mentor John Smith) as per the prototype.
    - [ ] 3c. There will be a 'Back' link that takes the user back to the previous screen in the journey.
    - [ ] 3d. When user clicks 'Confirm details' on the <ins>'Check your answers before registering [mentor name] as a mentor'</ins> page, mentor details will be saved and user will be navigated to the Confirmation page.
- [ ] 4. On the Confirmation page:
    - [ ] 4a. The mentor and ECT's name will be pulled into the confirmation message e.g. You've assigned [Samuel Taylor] as a mentor for John Smith.
    - [ ] 4b. There will be a link to go 'Back to your ECTs', which navigates the user to the 'View list of ECTs at your school' page (https://github.com/DFE-Digital/register-early-career-teachers/issues/468)
    - [ ] 4c. There will be no 'Back' link on the Confirmation page.
- [ ] 5. Session data should be destroyed at the start of the 'add mentor' journey.

## Technical notes
Note, we'll need to store the ECT's name from the start so we can play it back at the end.

## Linked tickets
Dependent on #758 #759 
Similar to #475 #478

**Comment by ltello** [comment]
[Review app](https://cpd-ec2-review-886-web.test.teacherservices.cloud/)

**Comment by gracecbrennan** [comment]
Toby has left comments. Some of which relate to approaches for build. 
Would be good for @peteryates to come in to help getting to a decision

**Comment by gracecbrennan** [comment]
Lorenzo needs to make a few more edits before moving to product review

**Comment by JSJohal** [comment]
Have reviewed and ticked off checklist - looks good to me.
